Dockerfile

```diff
@@ -19,156 +20,330 @@
+
+ARG BOWER_VERSION=1.8.8
+ARG NODEJS_VERSION=16.13.2
+ARG NPM_VERSION=7.20.6
+ARG NVM_DIR=/opt/nodejs
+ARG YARN_VERSION=1.22.10
+
+RUN git clone --depth 1 https://github.com/nvm-sh/nvm.git $NVM_DIR
+RUN . $NVM_DIR/nvm.sh \
+    && nvm install "${NODEJS_VERSION}" \
+    && nvm alias default "${NODEJS_VERSION}" \
+    && nvm use default \
+    && npm install --global npm@$NPM_VERSION bower@$BOWER_VERSION yarn@$YARN_VERSION
+
+#------------------------------------------------------------------------
+# GOLANG - Build as a separate component
+FROM build AS gobuild
+
+ARG GO_DEP_VERSION=0.5.4
+ARG GO_VERSION=1.17.3
+ENV GOPATH=/opt/go
+RUN curl -L https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz | tar -C /opt -xz
+ENV PATH=/opt/go/bin:$PATH
+RUN go version
+RUN curl -ksS https://raw.githubusercontent.com/golang/dep/v$GO_DEP_VERSION/install.sh | bash
+RUN echo "add_local_path /opt/go/bin:\$PATH" > /etc/profile.d/go.sh
+
+#------------------------------------------------------------------------
+# HASKELL STACK
+FROM build AS haskellbuild
+
+ARG HASKELL_STACK_VERSION=2.1.3
+RUN curl -ksS https://raw.githubusercontent.com/commercialhaskell/stack/v$HASKELL_STACK_VERSION/etc/scripts/get-stack.sh | bash -s -- -d /usr/bin
+
+#------------------------------------------------------------------------
+# REPO / ANDROID SDK
+FROM build AS androidbuild
+
+ARG ANDROID_CMD_VERSION=7583922
+ENV ANDROID_HOME=/opt/android-sdk
+
+RUN curl -ksS https://storage.googleapis.com/git-repo-downloads/repo > /usr/bin/repo \
+    && chmod a+x /usr/bin/repo
+
+RUN curl -Os https://dl.google.com/android/repository/commandlinetools-linux-${ANDROID_CMD_VERSION}_latest.zip \
+    && unzip -q commandlinetools-linux-${ANDROID_CMD_VERSION}_latest.zip -d $ANDROID_HOME \
+    && rm commandlinetools-linux-${ANDROID_CMD_VERSION}_latest.zip \
+    && PROXY_HOST_AND_PORT=${https_proxy#*://} \
+    && if [ -n "$PROXY_HOST_AND_PORT" ]; then \
+    # While sdkmanager uses HTTPS by default, the proxy type is still called "http".
+    SDK_MANAGER_PROXY_OPTIONS="--proxy=http --proxy_host=${PROXY_HOST_AND_PORT%:*} --proxy_port=${PROXY_HOST_AND_PORT##*:}"; \
+    fi \
+    && yes | $ANDROID_HOME/cmdline-tools/bin/sdkmanager $SDK_MANAGER_PROXY_OPTIONS \
+    --sdk_root=$ANDROID_HOME "platform-tools" "cmdline-tools;latest"
+COPY docker/android.sh /etc/profile.d
+
+#------------------------------------------------------------------------
+# SCANCODE - Build ScanCode as a separate component
+FROM pythonbuild AS scancodebuild
 
-# Set this to the ScanCode version to use.
 ARG SCANCODE_VERSION="30.1.0"
+ENV SCANCODE_URL "https://github.com/nexB/scancode-toolkit/archive/refs/tags/"
+ENV PATH=/opt/python/shims:$PATH
 
-FROM adoptopenjdk:11-jdk-hotspot-focal AS build
+RUN mkdir -p /opt/scancode \
+    && curl -ksSL ${SCANCODE_URL}/v${SCANCODE_VERSION}.tar.gz | tar -C /opt/scancode -xz --strip-components=1 \
+    && cd /opt/scancode \
+    && PYTHON_EXE=python3 ./configure \
```

Comment:
I guess we should now also install this via `pip` like done in the `main` branch.

Reply:
I would neither configure it yourself, nor use pip:

I think their preferred way of installing is "download and extract the release from github" https://github.com/nexB/scancode-toolkit/blob/develop/docs/source/getting-started/install.rst#installation-on-linux-and-mac

pip is just under the category "Installation as a library", so I would assume it is not the recommended way for usage.

Reply:
For now we go with PIP. 
There's no guarantee that the binary will be 100% statically compatible ( yes, this kind of thing exists on Linux )
So, pip is good enough. After we merge the PR, we discuss with Phillipe properly